src/tree/gpu_hist/row_partitioner.cu

```diff
@@ -32,22 +34,30 @@ void RowPartitioner::SortPosition(common::Span<TreePositionT> position,
   auto d_ridx_out = ridx_out.data();
   auto d_ridx_in = ridx.data();
   auto write_results = [=] __device__(size_t idx, int ex_scan_result) {
+    // the ex_scan_result represents how many rows have been assigned to left node so far
+    // during scan.
```

Comment:
This is a radix sort over 1 bit described in this article: https://developer.nvidia.com/gpugems/GPUGems3/gpugems3_ch39.html